extensions/npm/src/tasks.ts

```diff
@@ -104,49 +103,61 @@ function isTestTask(name: string): boolean {
 	}
 	return false;
 }
+const preScripts: Set<string> = new Set([
+	'est', /* test typo? */ 'install', 'pack', 'pack', 'publish', 'restart',
+	'shrinkwrap', 'stop', 'uninstall', 'version'
+]);
+
+const postScripts: Set<string> = new Set([
+	'install', 'pack', 'pack', 'publish', 'publishOnly', 'restart', 'shrinkwrap',
+	'stop', 'test', 'uninstall', 'version'
+]);
+
+function canHavePrePostScript(name: string): boolean {
+	return preScripts.has(name) || postScripts.has(name);
+}
 
-function isPrePostScript(name: string): boolean {
-	const prePostScripts: Set<string> = new Set([
-		'preuninstall', 'postuninstall', 'prepack', 'postpack', 'preinstall', 'postinstall',
-		'prepack', 'postpack', 'prepublish', 'postpublish', 'preversion', 'postversion',
-		'prestop', 'poststop', 'prerestart', 'postrestart', 'preshrinkwrap', 'postshrinkwrap',
-		'pretest', 'postest', 'prepublishOnly'
-	]);
+export function isWorkspaceFolder(value: any): value is WorkspaceFolder {
+	return value && typeof value !== 'number';
+}
 
-	const prepost = ['pre' + name, 'post' + name];
-	for (const knownScript of prePostScripts) {
-		if (knownScript === prepost[0] || knownScript === prepost[1]) {
-			return true;
-		}
+export async function getScriptRunner(folder: Uri, context?: ExtensionContext, showWarning?: boolean): Promise<string> {
+	let scriptRunner = workspace.getConfiguration('npm', folder).get<string>('scriptRunner', 'npm');
```

Comment:
I think the default in `get` here should be `auto`, and if it's auto then we `return getPackageManager(...)`

Reply:
`getPackageManager` also defaults to `auto` and call `detectPackageManager`. They basically do the same thing, just get different configuration.